src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```
@@ -0,0 +1,795 @@
+            var timeToSleep = TimeSpan.FromSeconds(1);
+
+            //Waiting for finish draining all the ownership leases and release them. 
+            while (!isFinish)
+            {
+                try
+                {
+                    isFinish = await this.tableLeaseManager.ShutDown();
+                }
+                //If the worker failed to update the table due to concurrency, re-read the table immediately without wait.
+                catch (RequestFailedException ex) when (ex.Status == 412)
+                {
+                    timeToSleep = TimeSpan.FromSeconds(0);
+                }
+                //Eat any other exceptions.
+                catch(Exception exception)
+                {
+                    this.settings.Logger.PartitionManagerError(
+                        this.storageAccountName,
+                        this.settings.TaskHubName,
+                        this.settings.WorkerId,
+                        "",
+                        $"Error in stopping partition manager : {exception}");
+                }
+                await Task.Delay(timeToSleep);
+            };
+
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "",
+                $"Successfully released all ownership leases for shutdown.");
+        }
+
+        async Task IPartitionManager.CreateLeaseStore()
+        {
+            try
+            {
+                await this.partitionTable.CreateIfNotExistsAsync();
+            }
+            catch (Exception exception)
+            {
+                this.settings.Logger.PartitionManagerError(
+                    this.storageAccountName,
+                    this.settings.TaskHubName,
+                    this.settings.WorkerId,
+                    "",
+                    $"Failed to create partition table: {exception}");
+            }
```

Is it safe for us to swallow this exception? Normally we would not swallow this kind of initialization exception unless its expected and we can operate normally even if it fails. I'm not sure that's the case here.

---

Oh, my initial thought is that if a worker failed to create the table/ partitions, maybe another worker could create it successfully because we usually have multiple workers. And this worker can still manage the table. I don't want it to be stuck here.  Otherwise, we have to restart / remove the role instance. But my assumption might be wrong. What do you think of it?

---

It's best to not rely on other workers to complete this work. For example, if there's only a single worker, then the table creation will not be retried and the app will be stuck. I recommend *not* swallowing this exception. That way, the app can immediately restart and try again until it succeeds.